datajunction-server/datajunction_server/construction/build_v2.py

```diff
@@ -589,6 +608,20 @@ async def build(self) -> ast.Query:
                     order=order,
                 )
 
+        # Replace any parameters in the final AST with their values
+        for param in self.final_ast.find_all(ast.QueryParameter):  # type: ignore
+            if param.name in self.parameters and param.parent:
+                param.parent.replace(param, self.parameters[param.name])
+            else:
+                raise DJQueryBuildException(
+                    errors=[
+                        DJQueryBuildError(
+                            code=ErrorCode.MISSING_PARAMETER,
+                            message=f"Missing parameter: {param.name}",
```

Comment:
`s/Missing parameter/Missing value for parameter`

Reply:
good idea, done